src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        // Config values pulled from CoreCLR
+        private static readonly HillClimbing s_threadPoolHillClimber = new HillClimbing(4, 20, 100, 8, 15, 300, 4, 20, 10, 200, 1, 200, 15);
+
+        private class HillClimbing
+        {
+            public enum State
```

Comment:
This needs to be the same accessibility as `ForceChange` below, which is public. This isn't actually used to track internal state. It's only used in logging states (which is currently TODO stubbed out).